Tribler/community/gigachannel/community.py

```diff
@@ -89,13 +104,64 @@ def on_blob(self, peer, blob):
         if reply_blob:
             self.endpoint.send(peer.address, self.ezr_pack(self.NEWS_PUSH_MESSAGE, RawBlobPayload(reply_blob)))
 
+    def send_search_request(self, query_filter, metadata_type='', sort_by="HEALTH", sort_asc=0, hide_xxx=True):
+        self.search_counter += 1
+        request = SearchRequestPayload(self.search_counter, query_filter.encode('utf8'),
+                                       metadata_type, sort_by, sort_asc, hide_xxx)
+        self._logger.debug("Started remote search for query:%s", query_filter)
+
+        for peer in self.get_peers()[:max_search_peers]:
+            self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_REQUEST, request))
+
+    @lazy_wrapper(SearchRequestPayload)
+    def on_search_request(self, peer, request):
+        query_filter = request.query_filter.decode('utf8')
+        metadata_type = {'': [REGULAR_TORRENT, CHANNEL_TORRENT],
+                         "channel": CHANNEL_TORRENT,
+                         "torrent": REGULAR_TORRENT
+                        }.get(request.metadata_type, REGULAR_TORRENT)
+
+        request_dict = {
+            "first": 1,
+            "last": max_entries,
+            "sort_by": request.sort_by,
+            "sort_asc": request.sort_asc,
+            "query_filter": query_filter,
+            "hide_xxx": request.hide_xxx,
+            "metadata_type": metadata_type,
+            "exclude_legacy": True
+        }
+
+        result_blob = None
+        with db_session:
+            db_results, total = self.metadata_store.TorrentMetadata.get_entries(**request_dict)
+            if total > 0:
+                result_blob = entries_to_chunk(db_results[:max_entries], maximum_payload_size)[0]
+        if result_blob:
+            self.endpoint.send(peer.address, self.ezr_pack(self.SEARCH_RESPONSE,
+                                                           SearchResponsePayload(request.id, result_blob)))
+
+    @lazy_wrapper(SearchResponsePayload)
+    def on_search_response(self, peer, response):
+        if self.search_counter != response.id:
+            return
+        with db_session:
+            try:
+                metadata_result = self.metadata_store.process_compressed_mdblob(response.raw_blob)
+            except (TransactionIntegrityError, CacheIndexError) as err:
+                self._logger.error("DB transaction error when tried to process search payload: %s", str(err))
+                return
+
+            search_results = [(dict(type={REGULAR_TORRENT: 'torrent', CHANNEL_TORRENT: 'channel'}[r.metadata_type],
```

Comment:
`process_compressed_mdblob` can have other types of return values. The interpretation of these values is important. For example, it can respond with a `GOT_NEWER_VERSION` and a corresponding DB object. So, you have to check the resulting action type and act properly:
1. Only notify on `UPDATED_OUR_VERSION` or `UNKNOWN_CHANNEL'/`UNKNOWN_TORRENT`.
2. If there is `GOT_NEWER_VERSION` response, you should send back the packet with the updated version, as it is done in the gossip protocol.

Reply:
Do we want this for this release? I think we could optimize this later.

Reply:
@xoriole , it is not about optimization. It is about other peers not being able to crash our Tribler instance with an unhandled response. And it is really easy to add response packet sent on `GOT_NEWER_VERSION`. Just copy it from gossip message handling in the same community.